### PR TITLE
refactor/parsec: simplify adding accusations and sync events.

### PR DESCRIPTION
**Pull request by jeanphilippeD**
We will add a new kind of event to add at the end with DkgMessages.
Clarify how we are doing it with our existing events.

### CODE REVIEW

File: src/parsec.rs [hunk 1]
```diff
@@ -1554,9 +1556,10 @@ impl<T: NetworkEvent, S: SecretId> Parsec<T, S> {
             )?;
 
             let _ = self.add_event(event)?;
-            Ok(None)
+            Ok(())
```
Comment (maqi) [inline src/parsec.rs]: The both branch return with `Ok(())`, can we make the function not requires returning with `Result`?
or at least move `Ok(())` out of the if clause to save one line?

Comment (jeanphilippeD) [reply src/parsec.rs]: There are `?` in the code, so Result is necessary.

Comment (maqi) [reply src/parsec.rs]: Then at least move `Ok(())` out of the if-clause ?

File: src/parsec.rs [hunk 2]
```diff
@@ -1445,28 +1440,42 @@ impl<T: NetworkEvent, S: SecretId> Parsec<T, S> {
...
-                } => self.add_sync_event(is_request, other_parent)?,
-                #[cfg(feature = "malice-detection")]
-                PendingEvent::Accusation {
-                    offender,
-                    malice,
-                    other_parent,
-                } => {
-                    let _ = self
-                        .add_accusation_event(offender, malice, other_parent)?
-                        .map(|unhandled| self.pending_accusations.push((offender, unhandled)));
```
Comment (pierrechevalier83) [inline src/parsec.rs]: I don't know for sure what it entails, but I think this behaviour of pushing unhandled accusations to the pending_accusations got lost in the refactoring, which makes it more than a pure refactoring.
Am I missing something?

Comment (jeanphilippeD) [reply src/parsec.rs]: No, it got moved inside add_accusation_event (unless I got it wrong in another way).

**Pull request merged**